### PR TITLE
Create Endpoint: POST to favorites

### PR DESCRIPTION
**Pull request by fentontaylor**
### Summary
For the endpoint `POST /api/v1/favorites`
1. First the JSON body is checked for required attributes `title`, `artistName`, `rating`.
    - If attribute is missing, return status 400 and specific error message with name of missing attribute.
2. Check the rating attribute to verify it is an integer between 1-100.
    - If not, return status 400 with error message.
3. If JSON has all valid attributes, create a new Favorite object to format the attributes
4. Insert Favorite object into database.

### Also Includes
- Happy and sad path tests for POST endpoint
- Test for Favorite model with/without given genre

### CODE REVIEW

File: models/favorite.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Favorite {
+  constructor(obj) {
+    this.title = obj.title,
+    this.artist_name = obj.artistName,
+    this.genre = obj.genre || 'Unknown',
```
Comment (nathangthomas) [inline models/favorite.js]: I do like the JS syntax for this much better than the Ruby equivalent.

File: tests/favorites/create.spec.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    expect(res.statusCode).toBe(201);
+
+    const fav = await database('favorites').first();
+    expect(fav.title).toBe(body.title);
+    expect(fav.artist_name).toBe(body.artistName);
+    expect(fav.genre).toBe(body.genre);
+    expect(fav.rating).toBe(body.rating);
+  })
+
+  it("happy path with default genre", async ()=> {
```
Comment (nathangthomas) [inline tests/favorites/create.spec.js]: Good thinking to test for this as well.

File: tests/favorites/create.spec.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    };
+
+    var res = await request(app)
+      .post("/api/v1/favorites")
+      .send(body);
+
+    expect(res.statusCode).toBe(400)
+    expect(res.body).toMatchObject({ error: "Rating must be between 1-100" })
+  })
+})
```
Comment (nathangthomas) [inline tests/favorites/create.spec.js]: Very thorough testing!

File: models/favorite.js (hunk 1 above)
Comment (fentontaylor) [reply models/favorite.js]: You can actually do the same thing in Ruby when assigning instance variables: `@genre = obj['genre'] || 'Unknown'`, but I do agree that the `.genre` looks cleaner.

**Comment by fentontaylor** [comment]
> Don't forget to add to the ReadMe as we go.
> Also, I pulled down your branch and am getting the following error in create.spec.js:
> ` error: TRUNCATE TABLE favorites CASCADE - relation "favorites" does not existerror: relation "favorites" does not exist` I haven't gotten a chase to dig very deep on it as I am about to head out the door. Considering we will be together this morning I think we should wait to merge until we can look at this together. I'm sure it is on my end, but just in case.

Good catch. Working on the README now. That error you are getting sounds like you didn't run the migration for the test environment: `knex migrate:latest --env test`

File: README.md [hunk 4]
```diff
@@ -1,74 +1,119 @@
...
+2) Next, you'll need to set up the database. In the terminal, run:
 ```
 psql
-CREATE DATABASE DATABASE_NAME_dev;
+CREATE DATABASE play_dev;
+CREATE DATABASE play_test;
 \q
 
 knex migrate:latest
+knex migrate:latest --env test
```
Comment (nathangthomas) [inline README.md]: This looks great! You were right about me having forgot to run migrations for the test environment. That also solves my testing issues. Doh!

**Pull request merged**